### PR TITLE
[nnyeah] Add remapping of non-ctor invocations of IntPtr ctor of NSObjects

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: tools/nnyeah/nnyeah/ConstructorTransforms.cs [hunk 1]
```diff
@@ -173,23 +180,51 @@ void ReworkCodeBlockAsNeeded (IEnumerable<MethodDefinition> methods)
...
 		}
+
+		public bool TryGetConstructorCallTransformation (Instruction instruction, [NotNullWhen (returnValue: true)] out Transformation? result)
+		{
+			if (instruction.OpCode.Code == Mono.Cecil.Cil.Code.Newobj &&
+				instruction.Operand is MethodDefinition invokedMethod &&
+				TransformedConstructors.Contains (invokedMethod)) {
+				switch (invokedMethod.Parameters.Count) {
+				case 1:
+					result = new Transformation (null!, TransformationAction.Insert, new List<Instruction> () {
```
Comment (mandel-macaque) [inline tools/nnyeah/nnyeah/ConstructorTransforms.cs:210]: If we are going to a low the transformation to take null, should we just make it nullable? Else the code in the transformation class is going to be written assuming it cannot be null when that is a lie.

File: tools/nnyeah/tests/unit/utils/ReworkerHelper.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		}
+
+		public static async Task<TypeDefinition> CompileTypeForTest (string code, string typeName = "Foo")
+		{
+			string lib = await TestRunning.BuildTemporaryLibrary (code);
+			var module = ModuleDefinition.ReadModule (lib, ReaderParameters);
+			return module.GetType (typeName);
+		}
+
+        public static Reworker? CreateReworker (string modulePath)
```
Comment (mandel-macaque) [inline tools/nnyeah/tests/unit/utils/ReworkerHelper.cs:31]: Funky indentation.

File: tools/nnyeah/tests/unit/utils/ReworkerHelper.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		{
+			var reworker = CreateReworker (inModule);
+			if (reworker is null)
+				return null;
+
+			var outStm = new MemoryStream ();
+			reworker.Rework (outStm);
+			outStm.Seek (0, SeekOrigin.Begin);
+			return ModuleDefinition.ReadModule (outStm, ReaderParameters);
+		}
```
Comment (mandel-macaque) [inline tools/nnyeah/tests/unit/utils/ReworkerHelper.cs:56]: Not for this PR, but I have see the GetReworkeModule implementation in several PRs and diff test fixtures, we probably can generalise it.

**Review by mandel-macaque — APPROVED** [review]
Please fix the indentation, considering the nullability of Transformation and maybe in a diff PR we can deal with the code repetition in tests.

**Comment by chamons** [comment]
I'm looking at fixing Manuel's comments this morning.

**Comment by chamons** [comment]
Steve fixed some of these concerns in a redo in https://github.com/xamarin/xamarin-macios/pull/15246

**Pull request closed without merging**